datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -141,7 +141,11 @@ impl PhysicalExpr for ScalarFunctionExpr {
             .collect::<Result<Vec<_>>>()?;
 
         // evaluate the function
-        let output = self.fun.invoke_batch(&inputs, batch.num_rows())?;
+        let output = self.fun.invoke_with_args(ScalarFunctionArgs {
```

Comment:
If we can pass in `inputs` directly here (not as a slice but as the Vec) that would be ideal

Reply:
I am a little confused, previously invoke took `invoke(&self, args: &[ColumnarValue])` so this is what is passed to `ScalarFunctionArgs { args: &[ColumnarValue], ...}`, would you like this changed to `&Vec<ColumnarValue>`?